### ISSUE TITLE
[transform/ExtractPatterns] undefined function "ExtractPatterns"

### ISSUE THREAD

**Issue by rloginovskiy** [open-telemetry/opentelemetry-collector-contrib#26102]
### Component(s)

processor/transform

## Description

Collector failing to start when using transform processor with ExtractPatterns function

## Steps to Reproduce

1. Define ExtractPatterns
2. Apply config 

Collector version: 0.83.0

## Expected Result

No error is thrown

## Actual Result

Error is thrown

### Collector version

0.83.0

## Environment
Kubernetes v1.25.4
Collector image: "otel/opentelemetry-collector-contrib:0.83.0"

### OpenTelemetry Collector configuration

```yaml
exporters:
      logging: {}
      otlp:
        endpoint: 'opentelemetry-demo-jaeger-collector:4317'
        tls:
          insecure: true
      prometheus:
        enable_open_metrics: true
        endpoint: 0.0.0.0:9464
        resource_to_telemetry_conversion:
          enabled: true
      coralogix:
        # The Coralogix domain
        domain: "coralogix.com"
        # Your Coralogix private key is sensitive
        private_key: ""
        application_name_attributes:
        - "k8s.namespace.name" 
        - "service.namespace"
        subsystem_name_attributes:
        - "k8s.deployment.name"
        - "k8s.statefulset.name"
        - "k8s.daemonset.name"
        - "k8s.cronjob.name"
        - "k8s.job.name"
        - "k8s.container.name"
        - "k8s.node.name"
        - "service.name"
        application_name: "otel-col"
        subsystem_name: "web"

        # (Optional) Timeout is the timeout for every attempt to send data to the backend.
        timeout: 30s
        traces:
          endpoint: "ingress.coralogix.com:443"
        metrics:
          endpoint: "ingress.coralogix.com:443"
        logs:
          endpoint: "ingress.coralogix.com:443" 
    extensions:
      health_check: {}
      memory_ballast:
        size_in_percentage: 40
    processors:
      batch: {}
      filter/ottl:
        error_mode: ignore
        metrics:
          metric:
          - name == "queueSize"
      k8sattributes:
        filter:
          node_from_env_var: KUBE_NODE_NAME     
        extract:
          metadata:
            - "k8s.namespace.name"
            - "k8s.deployment.name"
            - "k8s.statefulset.name"
            - "k8s.daemonset.name"
            - "k8s.cronjob.name"
            - "k8s.job.name"
            - "k8s.pod.name"
            - "k8s.node.name"
      transform:
        error_mode: ignore
        log_statements:
          - context: log
            statements:
              - set(cache, ExtractPatterns(body, "\\[(?P<ids>.*)\\])"))
              - set(body, cache["ids"]) 
      memory_limiter:
        check_interval: 5s
        limit_percentage: 80
        spike_limit_percentage: 25
    receivers:
      filelog:
          include:
            - /var/log/pods/*/*/*.log
      jaeger:
        protocols:
          grpc:
            endpoint: ${env:MY_POD_IP}:14250
          thrift_compact:
            endpoint: ${env:MY_POD_IP}:6831
          thrift_http:
            endpoint: ${env:MY_POD_IP}:14268
      k8sobjects:
       objects:
         - name: events
           mode: watch
           group: events.k8s.io
      otlp:
        protocols:
          grpc:
            endpoint: ${env:MY_POD_IP}:4317
          http:
            cors:
              allowed_origins:
              - http://*
              - https://*
            endpoint: 0.0.0.0:4318
      prometheus:
        config:
          scrape_configs:
          - job_name: opentelemetry-collector
            scrape_interval: 10s
            static_configs:
            - targets:
              - ${env:MY_POD_IP}:8888
      zipkin:
        endpoint: ${env:MY_POD_IP}:9411
    connectors:
      spanmetrics: 
    service:
      extensions:
      - health_check
      - memory_ballast    
      pipelines:
        logs:
          exporters:
          - logging
          - coralogix
          processors:
          - memory_limiter
          - batch
          - k8sattributes
          - transform
          receivers:
          - otlp
          - k8sobjects
        metrics:
          exporters:
          - prometheus
          - logging
          - coralogix
          processors:
          - memory_limiter
          - filter/ottl
          - k8sattributes
          - transform
          - batch
          receivers:
          - otlp
          - spanmetrics
        traces:
          exporters:
          - otlp
          - logging
          - spanmetrics
          - coralogix
          processors:
          - memory_limiter
          - k8sattributes
          - batch
          receivers:
          - otlp
          - jaeger
          - zipkin        
      telemetry:
        metrics:
          address: ${env:MY_POD_IP}:8888
        logs:
           level: debug
```

### Log output

```shell
2023-08-25 12:29:06 Error: invalid configuration: processors::transform: unable to parse OTTL statement "set(cache, ExtractPatterns(body, \"\\\\[(?P<ids>.*)\\\\])\"))": error while parsing arguments for call to "set": invalid argument at position 1: undefined function "ExtractPatterns"
2023-08-25 12:29:06 2023/08/25 10:29:06 collector server run finished with error: invalid configuration: processors::transform: unable to parse OTTL statement "set(cache, ExtractPatterns(body, \"\\\\[(?P<ids>.*)\\\\])\"))": error while parsing arguments for call to "set": invalid argument at position 1: undefined function "ExtractPatterns"
```

**Comment by TylerHelmuth** [comment]
@rloginovskiy The extract function has not been released yet. It will be available starting in 0.84.0.

**Comment by rloginovskiy** [comment]
Thank you for the information @TylerHelmuth !
